src/main/scala/org/constellation/genesis/Genesis.scala

```diff
@@ -23,156 +25,144 @@ object Genesis extends StrictLogging {
+                  coinbaseKey,
+                  normalized = false
+                )
             )
       )
-  }
 
-  private def createGenesisBlock(transactions: Seq[Transaction]): CheckpointBlock =
-    CheckpointBlock.createCheckpointBlock(transactions, GenesisTips)(CoinbaseKey)
+  private def createGenesisBlock(transactions: Seq[Transaction], coinbaseKey: KeyPair): CheckpointBlock =
+    CheckpointBlock.createCheckpointBlock(transactions, GenesisTips)(coinbaseKey)
 
   private def createEmptyBlockFromGenesis[F[_]: Sync: LiftIO](
-    genesisSOE: SignedObservationEdge
-  )(implicit dao: DAO): F[CheckpointBlock] = LiftIO[F].liftIO {
-    val dummyTransactionSrc = KeyUtils.makeKeyPair.getPublic.toId.address
-    val dummyTransactionDst = KeyUtils.makeKeyPair.getPublic.toId.address
-
-    dao.transactionService.createDummyTransaction(dummyTransactionSrc, dummyTransactionDst, CoinbaseKey).map { tx =>
-      CheckpointBlock.createCheckpointBlock(
-        Seq(tx),
-        Seq(
-          TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
-          TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
-        )
-      )(CoinbaseKey)
-    }
-  }
+    genesisSOE: SignedObservationEdge,
+    coinbaseKey: KeyPair
+  )(dao: DAO): F[CheckpointBlock] =
+    for {
+      dummyTransactionSrc <- Sync[F].delay(KeyUtils.makeKeyPair().getPublic.toId.address)
+      dummyTransactionDst <- Sync[F].delay(KeyUtils.makeKeyPair().getPublic.toId.address)
+      dummyTx <- LiftIO[F].liftIO {
+        dao.transactionService.createDummyTransaction(dummyTransactionSrc, dummyTransactionDst, coinbaseKey)
+      }
+      cb <- Sync[F].delay {
+        CheckpointBlock.createCheckpointBlock(
+          Seq(dummyTx),
+          Seq(
+            TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash)),
+            TypedEdgeHash(genesisSOE.hash, EdgeHashType.CheckpointHash, Some(genesisSOE.baseHash))
+          )
+        )(coinbaseKey)
+      }
+    } yield cb
 
   // TODO: Make F[Unit]
-  def createGenesisObservation(
+  def createGenesisObservation[F[_]: Sync: LiftIO](
     allocAccountBalances: Seq[AccountBalance] = Seq.empty
-  )(implicit dao: DAO): GenesisObservation = {
+  )(dao: DAO): F[GenesisObservation] =
     for {
-      genesis <- createDistributionTransactions[IO](allocAccountBalances)
-        .map(createGenesisBlock)
-      emptyA <- createEmptyBlockFromGenesis[IO](genesis.soe)
-      emptyB <- createEmptyBlockFromGenesis[IO](genesis.soe)
+      // TODO: That key should be provided externally by the genesis creator.
+      // We can use Node's KeyPair as well.
+      coinbaseKey <- Sync[F].delay(KeyUtils.makeKeyPair())
+      genesis <- createDistributionTransactions[F](allocAccountBalances, coinbaseKey)(dao)
+        .map(createGenesisBlock(_, coinbaseKey))
+      emptyA <- createEmptyBlockFromGenesis[F](genesis.soe, coinbaseKey)(dao)
```

Comment:
I think that `[F]` will be implicitly passed to `createEmptyBlockFromGenesis` and you don't need to explicitly point it out.